arangod/Agency/AgencyComm.cpp

```diff
@@ -810,66 +811,51 @@ uint64_t AgencyComm::uniqid(uint64_t count, double timeout) {
     return debugUniqId++;
   }
 #endif
-  static int const maxTries = 1000000;
-  // this is pretty much forever, but we simply cannot continue at all
-  // if we do not get a unique id from the agency.
-  int tries = 0;
+  AgencyCommResult readResult;
+  AgencyCommResult writeResult;
 
-  AgencyCommResult result;
+  TRI_ASSERT(!writeResult.successful());
 
   uint64_t oldValue = 0;
 
-  while (tries++ < maxTries) {
-    result = getValues("Sync/LatestID");
-    if (!result.successful()) {
-      std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  while (!writeResult.successful()) {
```

Comment:
@goedderz @neunhoef @maierlars:
The old code did check if the server went into shutdown mode.
I think we should also abort this (potentially endless) loop in case our own instance is shutting down.
As the caller expects an id back, the only way to abort this loop in this case would be by throwing an exception.
Additionally, in case the agency is not available for a long time, for whatever reason, do we really want to continue in this loop almost forever?

Reply:
> The old code did check if the server went into shutdown mode.
> I think we should also abort this (potentially endless) loop in case our own instance is shutting down.
> As the caller expects an id back, the only way to abort this loop in this case would be by throwing an exception.

Agreed, though I do not see where the old code did that. I'll add a check and an exception.

> Additionally, in case the agency is not available for a long time, for whatever reason, do we really want to continue in this loop almost forever?

I think yes: We will not be able to continue working in that state. Throwing an exception instead of waiting won't help that.

On a side note: I've arrived at the conclusion it'd be best to implement a more conservative backport of this PR for 3.7. Do you agree?